unified-runtime/source/adapters/opencl/usm.cpp

```diff
@@ -269,7 +269,8 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueUSMFill(
   // Have to look up the context from the kernel
   cl_context CLContext = hQueue->Context->CLContext;
 
-  if (patternSize <= 128 && isPowerOf2(patternSize)) {
+  if (patternSize <= 128 && isPowerOf2(patternSize) &&
+      isPointerAlignedTo(patternSize, &ptr)) {
```

Comment:
Is this right?  Seems like it should be:

```suggestion
      isPointerAlignedTo(patternSize, ptr)) {
```